src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -61,62 +165,83 @@ public Command shooterIdle() {
         .withName("Idle shooter command");
   }
 
+  public Command SetShooterIfReady(SpeakerConfig speeds, int exitCode) {
+    return runOnce(
+        () -> {
+          if (exitCode == 0) {
+            setShooterSpeeds(speeds);
+          }
+        });
+  }
```

Comment:
I think you want startEnd here so that the shooter speed gets set and then the Command stays scheduled even though it doesn't do anything actively until it ends. runOnce() will cause the default Command to take over 
```suggestion
    return startEnd(
        () -> {
          if (exitCode == 0) {
            setShooterSpeeds(speeds);
          }
        },
        () -> {});
  }
```